isvcs/zk.go

```diff
@@ -105,30 +100,50 @@ func initZK() {
 // This function sets up a ZooKeeper health check function parameterized with the connection string eg: "127.0.0.1:2181".
 func SetZKHealthCheck(connectString string) HealthCheckFunction {
 	return func(halt <-chan struct{}) error {
+		healthy := true
+		times := -1
 		for {
+			if !healthy && times == 3 {
+				log.WithFields(logrus.Fields{
```

Comment:
There's an easier way to do some of this logging, if you want to.

Instead of repeating with the `WithFields` in each call, you can add the following before line 102:
```
    logger := log.WithField("healthcheck", DEFAULT_HEALTHCHECK_NAME)
```
Then here at lines 107-109, you can do something like:
```
        logger.Warn("Unable to validate health of ZooKeeper. Retrying silently")
```

And wash/rinse/repeat for other logging calls in this method